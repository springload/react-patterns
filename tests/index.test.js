import React from "react";
import Patterns from "../src/index";
import renderer from "react-test-renderer";

it("renders correctly", async () => {
  // FIXME. CSS isn't working.
  global.document = {
    styleSheets: [
      {
        rules: ["b", "c"]
      }
    ]
  };

  const PatternBook = (
    <Patterns>
      <h1>test</h1>
    </Patterns>
  );

  const createNodeMock = elem => {
    return {
      innerHTML: "<h1>test</h1>",
      childNodes: ["a", "b"]
    };
  };
  const tree = renderer.create(PatternBook, { createNodeMock });

  const instance = tree.getInstance();
  // console.log(tree.toJSON().children[3].props.onClick());
  // console.log(tree.toJSON().children[4].props.onClick());

  const htmlButton = tree.toJSON().children[2].props;
  htmlButton.onClick();
  const cssButton = tree.toJSON().children[3].props;
  cssButton.onClick();
  // tree.props.onClick();

  expect(tree.toJSON()).toMatchSnapshot();
});
