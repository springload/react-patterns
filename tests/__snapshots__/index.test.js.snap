// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="react-patterns"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <button
    className="b-cart-toggle b-cart-toggle--not-in-cart"
    onClick={[Function]}
    title="Add to cart"
  >
    + Add to cart
  </button>
  <div
    className="react-patterns__example"
  >
    <h1>
      test
    </h1>
  </div>
  <details
    className="react-patterns__html"
    onClick={[Function]}
  >
    <summary>
      <abbr
        title="hypertext markup language"
      >
        HTML
      </abbr>
    </summary>
    <code
      className="b-code"
      dangerouslySetInnerHTML={
        Object {
          "__html": Array [],
        }
      }
    />
  </details>
  <details
    className="react-patterns__css"
    onClick={[Function]}
  >
    <summary>
      <abbr
        title="Cascading Style Sheets"
      >
        CSS
      </abbr>
    </summary>
    <code
      className="b-code"
      dangerouslySetInnerHTML={
        Object {
          "__html": Array [],
        }
      }
    />
  </details>
</div>
`;
